### PR TITLE
arrow-cpp: init at 0.8.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build arrow-cpp

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, symlinkJoin, fetchurl, boost, brotli, cmake, flatbuffers, gtest, gflags, lz4, rapidjson, snappy, zlib, zstd }:
+
+stdenv.mkDerivation rec {
+  name = "arrow-cpp-${version}";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "http://apache.mesi.com.ar/arrow/arrow-${version}/apache-arrow-${version}.tar.gz";
```
Comment (teh) [inline pkgs/development/libraries/arrow-cpp/default.nix]: Is this the best mirror?

File: pkgs/development/libraries/rapidjson/default.nix [hunk 2]
```diff
@@ -13,14 +13,16 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ pkgconfig cmake ];
 
-  # detected by gcc7
-  NIX_CFLAGS_COMPILE = [ "-Wno-error=implicit-fallthrough" ];
+  preConfigure = ''
+    substituteInPlace CMakeLists.txt --replace "-Werror" ""
```
Comment (teh) [inline pkgs/development/libraries/rapidjson/default.nix]: Can you add a comment why we need to disable errors completely?

Comment (veprbl) [reply pkgs/development/libraries/rapidjson/default.nix]: We could add
```-Wno-error=zero-as-null-pointer-constant``` and ```-Wno-error=shadow``` to use with modern clang, but what is the point? Compilers will get more sophisticated diagnostics, there will be more warnings and it's not like we actually going to fix any of the code. So we should just let developers compile with ```-Werror```

File: pkgs/development/libraries/arrow-cpp/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/libraries/arrow-cpp/default.nix]: changed to use ```mirror://```

**Comment by teh** [comment]
Hey, thanks for pulling this out!

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  RAPIDJSON_HOME = rapidjson;
+  SNAPPY_HOME = symlinkJoin { name="snappy-wrap"; paths = [ snappy snappy.dev ]; };
+  ZLIB_HOME = symlinkJoin { name="zlib-wrap"; paths = [ zlib.dev zlib.static ]; };
+  ZSTD_HOME = zstd;
+
+  meta = {
+    description = "A  cross-language development platform for in-memory data";
+    homepage = https://arrow.apache.org/;
+    license = stdenv.lib.licenses.asl20;
+    platforms = stdenv.lib.platforms.unix;
```
Comment (Mic92) [inline pkgs/development/libraries/arrow-cpp/default.nix]: Are you going to maintain this expression?

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow python35Packages.pyarrow python36Packages.pyarrow

**Comment by veprbl** [comment]
@grahamc build python2Packages.pyarrow

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, symlinkJoin, fetchurl, boost, brotli, cmake, flatbuffers, gtest, gflags, lz4, python, pythonPackages, rapidjson, snappy, zlib, zstd }:
```
Comment (FRidh) [inline pkgs/development/libraries/arrow-cpp/default.nix]: either python or pythonPackages

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow

**Comment by veprbl** [comment]
I can't reproduce darwin failure for pyarrow, it builds just fine for me on mac. This seems to be good to merge.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow

**Comment by veprbl** [comment]
@GrahamcOfBorg build python2Packages.pyarrow

**Comment by veprbl** [comment]
moved to #38440

**Pull request closed without merging**